frontend/src/tests/UsernameInput.spec.ts

```diff
@@ -0,0 +1,194 @@
+        const emitChange = wrapper.emitted('change');
+        expect(emitChange).toBeTruthy();
+        // test the given parameters when emitChange has been called
+        // i.e. emitChange = [ [ 'B' ] ]
+        expect(emitChange![0][0]).toEqual(newValue);
+
+        emitSetVerifyResult = wrapper.emitted('setVerifyResult');
+        await wrapper.setProps(newProps);
+        await verifyButton.trigger('click');
+
+        expect(emitSetVerifyResult).toBeTruthy();
+        expect(emitSetVerifyResult![1][0]).toEqual(true);
+    });
+
+    it('Should enable verify btn when username is inputted', async () => {
+        // button starts as disabled
+        expect(verifyButtonEl.disabled).toBe(true);
+        expect(verifyButton.classes('p-disabled')).toBe(true);
+
+        await wrapper.setProps({ userId: newProps.userId });
+
+        expect(verifyButtonEl.disabled).toBe(false);
+        expect(verifyButton.classes('p-disabled')).toBe(false);
+    });
+
+    it('Should show loading on the verify btn while we call the api', async () => {
+        vi.spyOn(
+            AppActlApiService.idirBceidProxyApi,
+            'idirSearch'
+        ).mockImplementation(async () => {
+            setLoadingState(true);
+            return userInputMock(true);
+        });
+
+        setLoadingState(false);
+        expect(verifyButtonEl.textContent).not.toContain('Loading');
+
+        await wrapper.setProps({ userId: newProps.userId });
+        await verifyButton.trigger('click');
+        expect(isLoading()).toBe(true);
+        expect(verifyButtonEl.textContent).toContain('Loading');
+
+        // cleanup state variable
+        setLoadingState(false);
+        expect(isLoading()).toBe(false);
+    });
+
+    it('Should show not found on card when user is not verified', async () => {
+        vi.spyOn(
+            AppActlApiService.idirBceidProxyApi,
+            'idirSearch'
+        ).mockImplementation(async () => {
+            return userInputMock(false);
+        });
+
+        await wrapper.setProps({ userId: newProps.userId });
+        await verifyButton.trigger('click');
+        await flushPromises();
+
+        userIdentity = wrapper.findComponent({
+            name: 'UserIdentityCard',
+        }).vm;
+
+        const cardEl = wrapper.find('.custom-card').element as HTMLSpanElement;
+        expect(cardEl).toBeTruthy();
+        expect(cardEl.textContent).toContain('User does not exist');
+    });
+
+    it('Should remove card and emit different value when domain changes', async () => {
+        let cardUsernameEl: HTMLSpanElement;
+        // default props
+        expect(wrapper.props()).toEqual(props);
+
+        await wrapper.setProps(userInputMock(true).data);
+        await verifyButton.trigger('click');
+        await flushPromises();
+
+        cardUsernameEl = wrapper.find('#userId').element as HTMLSpanElement;
+        expect(cardUsernameEl).toBeTruthy();
+        expect(cardUsernameEl.textContent).toContain('userId');
```

Comment:
let's add some comments here and do some cleanup, so no need to test the default props, and the whole test could just be:
```
         // show user identity card to prepare for the test 
         await wrapper.setProps({ userId: newProps.userId });
         await verifyButton.trigger('click');
         await flushPromises();
         const cardUsernameEl = wrapper.find('#userId').element as HTMLSpanElement;
         expect(cardUsernameEl).toBeTruthy();
         
         // change the domain to be B
         await wrapper.setProps({domain: UserType.B});
         // for BCeID should emit true
         expect(emitSetVerifyResult![1][0]).toEqual(true);
         // UserIdentityCard not on page anymore
         expect(wrapper.findAll('#UserIdentityCard')).toHaveLength(0);

         // change the domain to be I
         await wrapper.setProps({domain: UserType.I});
         // for IDIR should emit false
         expect(emitSetVerifyResult![1][0]).toEqual(false); 
```

Reply:
just a reminder to format the file again after the fixes, thanks so much!!

Reply:
I'll do that! Thank you!